### ISSUE TITLE
Compile requirements for E2E testsuite refactor

### ISSUE THREAD

**Issue by viccuad** [kubewarden/kubewarden-end-to-end-tests#112]
We would like to refactor `kubewarden-end-to-end-tests` to run them more and increase coverage. 

See https://github.com/kubewarden/rfc/blob/main/rfc/0004-tests-arch.md (simplistic),  https://github.com/kubewarden/kubewarden-end-to-end-tests/issues/68, https://github.com/kubewarden/kubewarden-end-to-end-tests/issues/22, https://github.com/kubewarden/kubewarden-end-to-end-tests/issues/30, https://github.com/kubewarden/kubewarden-controller/issues/134 for ideas and possible projects that can be reused.

## Acceptance criteria

- A new epic issue or RFC is opened.
- It lists all the **minimum** needed features, requirements and tests needed for an e2e testsuite that suits the several stakeholders and can help development (not only testing RCs).
  Some of these features are complimentary and inform greatly the architecture, for example being able to run airgap testscases at the same time we are able to test any helm-chart commit or any container-image tag. This unblocks the possibility of running the e2e tests against any PR and codebase if it is needed.
  
For example: 
- Types of testscases:
  - monitoring, audit-scanner, smokes, airgap, load tests, upgrade tests
  - approach used to run those testcases (some shipped with the helm chart at https://helm.sh/docs/topics/chart_tests/, running otel deps, running load testing tools, etc)
- Way of scheduling the testsuite to run:
  - CI and local runs
  - Possibility to use any helm-chart tgz or version
  - Possibility to run testcases in any order or not
  - Time and resources needed per e2e test run.
- List of nice-to-haves.
